server/db/models.py

```diff
@@ -107,6 +107,23 @@ def __hash__(self):
         return hash(self.name)
 
 
+class APIKeyTable(BaseModel):
+    __tablename__ = "api_keys"
+    id = Column(
+        UUIDType,
+        server_default=text("uuid_generate_v4()"),
+        primary_key=True,
+        index=True,
+    )
+    shop_id = Column(UUIDType, ForeignKey("shops.id"), nullable=False, index=True)
+    # keys are stored as a hash for security reasons
+    hashed_key = Column(String, nullable=False, unique=True, index=True)
+    created_at = Column(UtcTimestamp, server_default=text("CURRENT_TIMESTAMP"))
+    revoked_at = Column(UtcTimestamp, nullable=True)
+
+    shop = relationship("ShopTable", back_populates="api_keys")
```

Comment:
@acidjunk To what should API keys linked? A shop? (Initially I had it linked to a user, but that might create other problems..)

Reply:
As discussed, linked it to the cognito user id (not the old UserTable)